test/pg-cdc/pg-cdc.td

```diff
@@ -490,6 +490,24 @@ true
 > SELECT COUNT(*) > 0 FROM join_view;
 true
 
+#
+# Ensure we can start a source with more workers than the default max_wal_senders param (10)
+#
+
+> CREATE SOURCE large_cluster
+  FROM POSTGRES CONNECTION pg (PUBLICATION 'mz_source')
+  FOR TABLES ("pk_table" AS large_cluster_pk_table)
+  WITH (SIZE = '16');
+
+> SELECT * FROM large_cluster_pk_table;
+13 three
+14 four
+2 two_two
+5 five
+
+> SELECT status = 'starting' OR status = 'running' FROM mz_internal.mz_source_statuses WHERE name = 'large_cluster_pk_table';
```

Comment:
Should this not be _just_ `status = 'running'`? If it's stuck at starting that's a bug, no?

Reply:
Job security! #20674

Reply:
Sources used to go directly into the running state but then it was difficult to figure out if the ingestion was restarting by looking at the status history, so we made it such that we always set the status to `starting` and it only flips to `running` once we see one successful record, which means that if a table is empty it will get stuck in `starting`. We can make it such that on initialization we always set the status to starting and then immediately back to running.

Reply:
That sounds great. @sploiselle filed https://github.com/MaterializeInc/materialize/issues/20674 to track this.

Reply:
Well, I don't think we want to _immediately_ set it to running. Is it easy to wait to go from starting -> running until the connection to Postgres has been established?